src/Compilers/CSharp/Test/Emit2/Diagnostics/DiagnosticSuppressorTests.cs

```diff
@@ -70,6 +70,72 @@ void M(int i)
             VerifySuppressedAndFilteredDiagnostics(compilation, analyzers);
         }
 
+        [WorkItem(62540, "https://github.com/dotnet/roslyn/issues/62540")]
+        [Fact]
+        public void TestSuppression_CompilerSyntaxBindingError_AndSuppressibleWarning()
+        {
+            const string SourceCode = @"
+                public class MyClass
+                {
+                    void MyPrivateMethod(int i)
+                    {
+                        // warning CS1522: Empty switch block
+                        // NOTE: Empty switch block warning is reported by the C# language parser
+                        switch (i)
+                        {
+                        }
+                    }
+                }
+                public class YourClass
+                { 
+                    void YourPrivateMethod()
+                    {
+                        // Cannot access private method
+                        new MyClass().MyPrivateMethod();
+                    }
+                }";
+
+            var compilation = CreateCompilation(SourceCode);
+
+            // Verify that suppression takes place even there are declaration errors
+            var analyzers = new DiagnosticAnalyzer[] { new DiagnosticSuppressorForId("CS1522") };
+            VerifySuppressedDiagnostics(compilation, analyzers, Diagnostic("CS1522", "{", isSuppressed: true));
```

Comment:
Please include the generated comments for the `Diagnostic()` results like we do in `TestSuppression_CompilerSyntaxWarning`.